shared/src/main/scala/amf/core/utils/UriUtils.scala

```diff
@@ -0,0 +1,71 @@
+package amf.core.utils
+
+import amf.Core.platform
+
+object UriUtils {
+
+  def resolveRelativeTo(current: String, rawUrl: String): String = {
+    val base = rawUrl match {
+      case Absolute(_)               => None
+      case RelativeToIncludedFile(_) => Some(current)
+    }
+    val encodedUrl = platform.encodeURI(rawUrl)
+    resolve(base, encodedUrl)
+  }
+
+  def resolve(base: Option[String], url: String): String = {
+    val result = base.map { baseUri =>
+      if (url.startsWith("#")) baseUri + url
+      else {
+        val baseDir = stripFileName(baseUri)
+        safeConcat(baseDir, url)
+      }
+    }.getOrElse(url)
+    platform.resolvePath(result)
+  }
+
+  private def safeConcat(base: String, url: String) = {
+    if (base.nonEmpty && base.last == '/' && url.nonEmpty && url.head == '/') base + url.drop(1)
+    else if ((base == "file://") && url.startsWith("./")) base + url.substring(2) // associated to APIMF-2357
+    else base + url
+  }
+
+  def stripFileName(url: String): String = {
```

Comment:
we should have another method that receives the so, so we can test is separately